src/app/dev/DevToys.Tools/Tools/Converters/JsonTable/JsonTableConverterGuiTool.cs

```diff
@@ -0,0 +1,210 @@
+                                    .Language("json")
+                                    .OnTextChanged(OnInputTextChanged))
+                            .WithRightPaneChild(
+                                _outputDataGrid
+                                    .Title(JsonTableConverter.Output)
+                                    .CommandBarExtraContent(_copyButton)
+                                    .Extendable())
+                    )
+                )
+            );
+        }
+    }
+
+    // Smart detection handler.
+    public void OnDataReceived(string dataTypeName, object? parsedData)
+    {
+        if (dataTypeName == PredefinedCommonDataTypeNames.Json &&
+            parsedData is string jsonStrongTypedParsedData)
+        {
+            _inputTextArea.Text(jsonStrongTypedParsedData);
+        }
+    }
+
+    public void Dispose()
+    {
+        _cancellationTokenSource?.Cancel();
+        _cancellationTokenSource?.Dispose();
+        _semaphore.Dispose();
+    }
+
+    private void OnInputTextChanged(string text)
+    {
+        StartConvert(ConvertTarget.DataGrid);
+    }
+
+    private void StartConvert(ConvertTarget target)
+    {
+        _cancellationTokenSource?.Cancel();
+        _cancellationTokenSource?.Dispose();
+        _cancellationTokenSource = new CancellationTokenSource();
+
+        WorkTask = ConvertAsync(target, _cancellationTokenSource.Token);
+    }
+
+    private async Task ConvertAsync(ConvertTarget target, CancellationToken cancellationToken)
+    {
+        using (await _semaphore.WaitAsync(cancellationToken))
+        {
+            await TaskSchedulerAwaiter.SwitchOffMainThreadAsync(cancellationToken);
+
+            ConvertResult conversionResult = ConvertFromJson(_inputTextArea.Text, _settingsProvider.GetSetting(copyFormat), cancellationToken);
+
+            if (conversionResult.Error == ConvertResultError.None && conversionResult.Data != null)
+            {
+                _copyButton.Enable();
+                switch (target)
+                {
+                    case ConvertTarget.DataGrid:
+                        SetDataGridData(conversionResult.Data);
+                        break;
+
+                    case ConvertTarget.Clipboard:
+                        await _clipboard.SetClipboardTextAsync(conversionResult.Text);
+                        break;
+                }
+            }
+            else
+            {
+                _copyButton.Disable();
+                SetDataGridError();
+            }
+        }
+    }
+
+    private enum ConvertTarget { DataGrid, Clipboard }
+
+    private void SetDataGridData(DataGridContents table)
+    {
+        IUIDataGridRow[] rows = table.Rows.Select(r => Row(null, r)).ToArray();
+        _outputDataGrid.WithColumns(table.Headings);
+        _outputDataGrid.WithRows(rows);
+    }
+
+    private void SetDataGridError()
+    {
+        _outputDataGrid.WithColumns(JsonTableConverter.JsonError);
+        _outputDataGrid.WithRows();
```

Comment:
That's actually a pretty great idea!

Reply:
I copied the idea from the JSON>YAML tool which shows the error message in the output section :)